scripts/plugins/extauth-hook-AD.py

```diff
@@ -176,9 +176,22 @@ def _query_subject(self, sid):
         if len(comps) < 2:
             logger.error("subject %s from sid %s is not valid", subject, sid)
             return None, None
-        # CONNAPP\test_user 1  # 1: user, 2: group
-        name, category = comps[0].strip(), comps[1].strip()
-        return name, True if category == "2" else False
+        # CONNAPP\IUgroup 1  # 1: user, 2: group
+        is_group = comps[1].strip() == "2"
+        real_name = comps[0].strip()
+        # The real_name returned name contains uppercase, need to format it from subject details
+        return self._query_subject_name(real_name, is_group), is_group
+
+    def _query_subject_name(self, real_name, is_group):
+        sub_cmd = "--group-info" if is_group else "--user-info"
+        cmd = ["/usr/bin/wbinfo", sub_cmd, real_name]
+        subject_detail = run_cmd(cmd)
+        if not subject_detail:
+           logger.error("Failed to lookup subject details %s", real_name)
+           return None
+        # CONNAPP\ugroup:x:3000009: --> group details
+        # CONNAPP\iugroup:*:3000016:3000000:IUgroup:/home/CONNAPP/iugroup:/bin/bash --> user details
+        return subject_detail.split(":")[0]
```

Comment:
Will this also fix the case when we have a `#` in the group name?
That is a bug in the current PBIS code that groups starting with `#` (like many of our internal groups) do not work, because `#` is a comment character in PAM and cannot be escaped.
Are the names returned here guaranteed to be compatible with PAM and not contain characters like `#`?
Might be useful to add a testcase specifically for that.

Reply:
@edwintorok  I googled a while about how to escape a `#` for pam configuration (and quick go through the pam code, ),  but not found, 
is it a limitation of pam configuration?  any suggestion about this?

We probably can enum all the users in the group, but That sounds not a good idea.

Reply:
Looks like we would need to patch PAM to fix this. Lets consider it out of scope for this PR for now